app/main.py

```diff
@@ -1,8 +1,31 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name, age):
+        self.name = name
+        self.age = age
+        self.people[self.name, self.age] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people_dict: list) -> list:
+    person_list = []
+    for p_dict in people_dict:
+        p = Person(p_dict["name"], p_dict["age"])
+        p_dict["ss"] = p
+        person_list.append(p)
+    for p_dict in people_dict:
+        if "husband" in p_dict:
+            if p_dict["husband"] is not None:
```

Comment:
join two conditions in one `if`-statement.
nested constructions make code less readable